compiler/rustc_type_ir/src/canonical.rs

```diff
@@ -19,10 +19,31 @@ pub struct Canonical<I: Interner, V> {
     pub value: V,
     pub max_universe: UniverseIndex,
     // FIXME(lcnr, oli-obk): try moving this into the query inputs instead
-    pub defining_opaque_types: I::DefiningOpaqueTypes,
+    pub opaque_type_mode: OpaqueTypeMode<I>,
     pub variables: I::CanonicalVars,
 }
 
+#[derive(derivative::Derivative)]
+#[derivative(
+    Copy(bound = ""),
+    Clone(bound = ""),
+    Hash(bound = ""),
+    PartialEq(bound = ""),
+    Eq(bound = ""),
+    Debug(bound = "")
+)]
+#[cfg_attr(feature = "nightly", derive(TyEncodable, TyDecodable, HashStable_NoContext))]
+pub enum OpaqueTypeMode<I: Interner> {
+    Define(I::DefiningOpaqueTypes),
```

Comment:
the type alias should get renamed to `I::DefIdList` or sth 🤷

Reply:
Yeah